install_gridpack_deps.sh

```diff
@@ -71,62 +84,87 @@ then
   # Download, build, and install GA
   cd ${GP_EXT_DEPS}
 
-  echo "Downloading GA-5.8"
+  if [ "$install_shared" == "true" ]; then
+      gashopts="--enable-shared=yes --enable-static=no"
+  else
+      gashopts="--enable-shared=no --enable-static=yes"
+  fi
+
+  gaversion="5.8.2"
+  echo "Downloading GA-$gaversion"
 
-  wget https://github.com/GlobalArrays/ga/releases/download/v5.8/ga-5.8.tar.gz
+  wget "https://github.com/GlobalArrays/ga/releases/download/v${gaversion}/ga-${gaversion}.tar.gz"
 
-  tar -xf ga-5.8.tar.gz
+  tar -xf "ga-${gaversion}.tar.gz"
 
-  cd ga-5.8
+  cd ga-${gaversion}
 
   # Build GA
-  echo "Building GA-5.8"
-  ./configure --with-mpi-ts --disable-f77 --without-blas --enable-cxx --enable-i4 --prefix=${PWD}/install_for_gridpack --enable-shared
+  echo "Building GA-${gaversion}"
+  ./configure --with-mpi-ts --disable-f77 \
+              --without-blas --without-lapack --without-scalapack \
+              --enable-cxx --enable-i4 \
+              --prefix=${PWD}/install_for_gridpack \
+              CFLAGS=-Wno-implicit-function-declaration \
+              $gashopts
   
   # Install GA
-  echo "Installing GA-5.8"
+  echo "Installing GA-${gaversion}"
   make -j 10 install
   
-  echo "GA-5.8 installation complete"
+  echo "GA-${gaversion} installation complete"
 fi
 
 if ${install_petsc}
 then
+
+  if [ "$install_shared" == "true" ]; then
+      petscopts="--with-shared-libraries=1"
+  else
+      petscopts="--with-shared-libraries=0"
+  fi
+
+  # petsc_version="3.21.4"
+  petsc_version="3.20.6"
     
-  # Install PETSc 3.16.4
+  # Install PETSc 
   cd ${GP_EXT_DEPS}
 
+  rm -rf petsc
+
   # Download
-  echo "Downloading PETSc 3.16.4"
+  echo "Downloading PETSc $petsc_version"
 
   git clone https://gitlab.com/petsc/petsc.git
     
   cd petsc
 
-  git checkout tags/v3.16.4 -b v3.16.4
+  # git checkout tags/v3.16.4 -b v3.16.4
+  git checkout "tags/v$petsc_version" -b "v$petsc_version"
 
   export PETSC_DIR=${PWD}
   export PETSC_ARCH=build-dir
+  rm -rf $PETSC_ARCH
 
   # Install PETSc
-  echo "Installing PETSc 3.16.4"
+  echo "Installing PETSc $petsc_version"
     
-  ./configure --download-superlu_dist --download-metis --download-parmetis --download-suitesparse --download-f2cblaslapack --download-cmake --prefix=${PWD}/install_for_gridpack --scalar-type=complex --with-shared-libraries=1 --download-f2cblaslapack=1
+  ./configure --with-gnu-compilers=0 --download-superlu_dist --download-metis --download-parmetis --download-suitesparse --download-f2cblaslapack --download-cmake=0 --prefix=${PWD}/install_for_gridpack --scalar-type=complex  --download-sowing  --download-f2cblaslapack $petscopts
```

Comment:
- Why NO GNU compilers? 
- Scalar type should be real
- Also add `--download-scalapack --download-mumps`

Reply:
> * Why NO GNU compilers?

I should be able to remove this.  Some part of PETSc was choosing the wrong  GFortran.  I tried to use that option to resolve. It turned out to be unnecessary.  

>     * Scalar type should be real

OK, but this was not the case in `develop` when I started this branch.  

>     * Also add `--download-scalapack --download-mumps`

Agreed, but these were not the script from `develop` when I started this branch.  

I can do the latter two here, but this will mean a conflicts whenever the EMT stuff is merged.